contracts/FundsSplitter.sol

```diff
@@ -0,0 +1,54 @@
+pragma solidity ^0.5.0;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract FundsSplitter is Ownable {
+    using SafeMath for uint256;
+
+    address payable public platform;
+    address payable public partner;
+
+    uint256 public partnerRate = 7;
+
+    constructor (address payable _platform, address payable _partner) public {
+        platform = _platform;
+        partner = _partner;
+    }
+
+    function splitFunds(uint256 _totalPrice) internal {
+        if (msg.value > 0) {
+            uint256 refund = msg.value.sub(_totalPrice);
+
+            // overpaid...
+            if (refund > 0) {
+                msg.sender.transfer(refund);
+            }
+
+            // work out the amount to split and send it
+            uint256 partnerAmount = _totalPrice.div(100).mul(partnerRate);
```

Comment:
Do we need a check `if partnerRate > 0` - incase we set to zero at some point so it does burn the gas sending nothing

Reply:
happy with this for now